### ISSUE TITLE
gvim line height became noticeably bigger in 2.200

### ISSUE THREAD

**Issue by habamax** [JetBrains/JetBrainsMono#334]
## 2.200

![image](https://user-images.githubusercontent.com/234774/96743630-e3245f80-13cc-11eb-9a65-8a6f4502bee5.png)

## 2.002
![image](https://user-images.githubusercontent.com/234774/96744094-59c15d00-13cd-11eb-9bad-5e73258ab4a4.png)

**Comment by philippnurullin** [comment]
Looks like it is increased indeed. Here is discussion on how to manually make it smaller https://vi.stackexchange.com/questions/10043/how-to-change-the-space-between-lines-in-vim

**Comment by mgieseki** [comment]
The new line height also affects GVim's [airline](https://github.com/vim-airline/vim-airline) which is now higher so that the height of some special characters, like the left and right triangles, don't match any longer.

![airline](https://user-images.githubusercontent.com/3620467/96850966-bec19500-1457-11eb-88ba-c489163c9b1f.png)

**Comment by habamax** [comment]
Yep, I know how to workaround it in gvim with `set linespace=-3`. Was hoping this is font issue, not something 'by design'.

**Comment by philippnurullin** [comment]
It's difficult matter. There is 4 groups of metrics in font that can be used as reference points when calculation the line-height in the rendering process. The only metrics became bigger in the 2.200 is the `WinAscender`. It was increased to match the height the character with Vietnamese diacritics (this was a part of the process to meet Google Fonts criteria). So i assume that is why line height increased in Vim but this thought needs testing and understanding what is exact algorithm used in Vim. It seems like a complicated task so it will take a while.

**Comment by Zcowyzrg** [comment]
I can confirm the last usable font (for me) was the NL version 1.05:
![image](https://user-images.githubusercontent.com/51540128/96860484-b4ed6100-145a-11eb-917b-2f330eeb386a.png)
You can see nicely continuous vertical lines.

Unfortunately, the regular, non-NL version is worse:
![image](https://user-images.githubusercontent.com/51540128/96860649-e6fec300-145a-11eb-86e6-8bd5e171f010.png)

All the later versions from 1.06 and up to the current 2.200 are only getting worse, there's even more space between the lines.

I use the font on Windows with cmd.exe, Notepad++, Eclipse, putty - they all have the same issues with virtually no chance to correct it with settings.

PS. Kudos for the great font!

**Comment by philippnurullin** [comment]
@Zcowyzrg Well this problem actually can be fixed fairly easy. I'll put in the list for the next release.

**Comment by philippnurullin** [comment]
Ok, i pushed a demo fix. I will be grateful if you can test it.
1. Install `https://github.com/JetBrains/JetBrainsMono/blob/master/fonts/ttf/JetBrainsMonoDEV-Regular.ttf`
2. Select `JetBrains Mono DEV`
3. See if it bring any change in lihe-height. The Boxdrawing may be wonky. Didit corrected it to work with new line heights.

**Comment by mgieseki** [comment]
I can confirm this as well, and even the airline characters fit nicely together now.
Thanks for your great work and the neat font.

**Comment by philippnurullin** [comment]
Ok, Glad to hear! I'll include this fix in the next update then.

@mgieseki Can you share the screen with the Powerline example. I saw a nasty placement of triangles. Wonder if this if affected too?

**Comment by habamax** [comment]
@philippnurullin 
![image](https://user-images.githubusercontent.com/234774/97888847-ad706680-1d3c-11eb-8fd5-e557f56f6b66.png)

**Comment by mgieseki** [comment]
Here is another screenshot similar to my previous one above:

![airline2](https://user-images.githubusercontent.com/3620467/97889163-506cb480-1d2c-11eb-81fd-5f1fca6f8ad8.png)

**Comment by philippnurullin** [comment]
Thanks, looking better! I'll add a bit overlapping to eliminate the gaps.

**Comment by Zcowyzrg** [comment]
cmd.exe, NL-1.0.5:
![image](https://user-images.githubusercontent.com/51540128/97907581-659e0f00-1d3d-11eb-90a3-eee010b2dbb5.png)
cmd.exe, DEV (note weaker appearance, smaller font height, but the same line height)
![image](https://user-images.githubusercontent.com/51540128/97907660-849ca100-1d3d-11eb-88bc-b94ff16214e6.png)

Notepad++, NL-1.0.5, fits 42 lines in a window:
![image](https://user-images.githubusercontent.com/51540128/97908024-01c81600-1d3e-11eb-9c9d-60620e66d9f1.png)
Notepad++, DEV, fits 39 lines in the same window:
![image](https://user-images.githubusercontent.com/51540128/97908072-1ad0c700-1d3e-11eb-870f-68b43e6a77bb.png)

**Comment by shabahengam** [comment]
I have 40 line in vim with ver 2.200 and 37 line with DEV version.
Left: DEV version ... Right: ver 2.200
xterm, fontsize 10
![jet](https://user-images.githubusercontent.com/53096747/97933201-4e840f00-1d87-11eb-8991-f6ce4790c440.png)

**Comment by Zcowyzrg** [comment]
I have just checked the 2.200 font on my linux box - and I don't have any issues with the font like I do on Windows. Both the "normal" and the "NL" versions have identical line height, box drawing characters have no gaps, no "weaker appearance" (which I presume is a scaling issue in Windows). I couldn't find any relevant general font metric that could contribute to the issues in Windows - `ttfdump` shows differences in `head/xmin` value (NL:-118, normal:-1740), but no (read: none that I could identify) differences in y or line height. Also checked in FontForge - all the font metrics displayed in font information dialogs are identical.
What the heck is wrong with Windows? ;-)

**Comment by Virss1** [comment]
In VSCode on Windows 10 Pro (v2004), in version 2.200 symbols height became too high. But in 2.210 dev-version everything looks fine.
Here is comparison gif (at 200% size) with JavaScript code in VSCode at font size 13px of 4 versions of the font (1.0.2, 2.000, 2.200, 2.210):
![1 0 2, 2 000, 2 200, 2 210 (200%)](https://user-images.githubusercontent.com/44531490/98545420-47915b00-22b7-11eb-818a-b5b16260febf.gif)

**Comment by Noammac** [comment]
Pulled today and found that my terminal has become significantly larger than expected. I managed to fix it by manually specifying line height to be 2px shorter.

**Comment by N-R-K** [comment]
Updating from `2.001` to `2.242`, I see the same issue. I lost about 4 whole vertical lines on my terminal...
